sale_product_pack/models/sale_order.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def copy(self, default=None):
+        sale_copy = super().copy(default)
+        # we unlink pack lines that should not be copied
+        pack_copied_lines = sale_copy.order_line.filtered(
+            lambda l: l.pack_parent_line_id.order_id == self
+        )
+        pack_copied_lines.unlink()
+        return sale_copy
+
```

Comment:
```suggestion
    def write(self, vals):
        if 'order_line' in vals:
            to_delete_ids = [e[1] for e in vals['order_line'] if e[0] == 2]
            sub_packs = self.env['sale.order.line'].search(
                [('pack_parent_line_id', 'in', to_delete_ids)])
        res = super().write(vals)
        if sub_packs:
            sub_packs_command = [(2, c.id, 0) for c in sub_packs]
            self.write({'order_line': sub_packs_command})
        return res

```

Reply:
Yes, I would suggest something similar, but try to include in original `vals` the `[(2, ...)]` commands for doing all in once step.